cartographer/io/serialization_format_migration_test.cc

```diff
@@ -175,6 +234,41 @@ TEST_F(MigrationTest, SerializedDataOrderAfterGridMigrationIsCorrect) {
   EXPECT_TRUE(serialized[8].has_landmark_data());
 }
 
+TEST_F(SubmapHistogramMigrationTest,
+       SubmapHistogramGenerationFromTrajectoryNodes) {
+  mapping::MapById<mapping::SubmapId, mapping::proto::Submap>
+      submap_id_to_submap;
+  mapping::SubmapId submap_id(submap_.submap_id().trajectory_id(),
+                              submap_.submap_id().submap_index());
+  submap_id_to_submap.Insert(submap_id, submap_);
+
+  mapping::MapById<mapping::NodeId, mapping::proto::Node> node_id_to_node;
+  mapping::NodeId node_id(node_.node_id().trajectory_id(),
+                          node_.node_id().node_index());
+  node_id_to_node.Insert(node_id, node_);
+
+  const auto submap_id_to_submap_migrated =
+      MigrateSubmapFormatVersion1ToVersion2(submap_id_to_submap,
+                                            node_id_to_node, pose_graph_);
+
+  EXPECT_EQ(submap_id_to_submap_migrated.size(), submap_id_to_submap.size());
+  const mapping::proto::Submap& migrated_submap =
+      submap_id_to_submap_migrated.at(submap_id);
```

Comment:
Would the code from below this line be simpler if we just did something like:

mapping::proto::Submap expected_submap = submap_;
// then set the histogram field accordingly
EXPECT_THAT(migrated_submap, EqualsProto(expected_submap));

This would also test that no other fields got screwed up.
wdyt?

Reply:
`EqualsProto` seemed to only work on string formatted protos. (see [relative_pose_constraint_3d_test.cc](https://github.com/googlecartographer/cartographer/blob/master/cartographer/pose_graph/constraint/relative_pose_constraint_3d_test.cc))